catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergErrorMapper.java

```diff
@@ -89,6 +96,30 @@ public Response toResponse(Throwable ex, IcebergEntityKind kind) {
     return Response.status(code == null ? 500 : code).entity(body).build();
   }
 
+  private static String message(BackendErrorStatus status, Throwable ex) {
+    return String.format("%s (due to: %s)", ex.getMessage(), status.cause().toString());
+  }
+
+  private IcebergErrorResponse mapStorageFailure(BackendErrorStatus status, Throwable ex) {
+    // Log full stack trace on the server side for troubleshooting
+    LOGGER.info("Propagating storage failure to client: {}", status, ex);
+
+    switch (status.statusCode()) {
+      case THROTTLED:
+        return errorResponse(429, "RuntimeException", message(status, ex), ex);
```

Comment:
Does `type` need to be the name of an existing exception? I think `"TooManyRequests"` would be more explicit.

Reply:
It's inconsistent on the Iceberg client side, IIRC, but many call paths use java exception names, indeed.

Reply:
To add some complications: a bunch of Iceberg tests assert on the names

Reply:
(Not in this case though)

Reply:
Maybe put in `TooManyRequests` there in this case.

Reply:
I did not find any handlers for 429 in the REST Catalog client, so I guess it's fine to use any error `type` here. I changed to `TooManyRequestsException` to be in-line with the other cases.